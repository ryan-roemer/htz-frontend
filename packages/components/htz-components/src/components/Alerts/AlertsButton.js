import React, { Component, } from 'react';
import PropTypes from 'prop-types';
import AriaDescription from '../AriaDescription/AriaDescription';

const alertButtonPropTypes = {
  /**
   * The author object which the user can register for his alerts.
   */
  author: PropTypes.object.isRequired,
  /**
   * A function to execute when button is toggled.
   * (Use to show/hide the registration form)
   */
  onToggle: PropTypes.func,
};

const alertsButtonDefaultProps = {
  onToggle: null,
};

class AlertsButton extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isExpanded: false,
    };

    this.toggleExpanded = this.toggleExpanded.bind(this);
  }

  toggleExpanded() {
    this.setState(
      prevState => ({
        isExpanded: !prevState.isExpanded,
      }),
      () => {
        this.props.onToggle(this.state.isExpanded);
      }
    );
  }

  render() {
    const { author, className, children, forwardedRef, } = this.props;

    return (
      <button
        ref={forwardedRef}
        className={className}
        aria-expanded={this.state.isExpanded}
        onClick={this.toggleExpanded}
        aria-describedby="alerts_btn_description"
      >
        <AriaDescription id="alerts_btn_description">
          קבל התראות בתיבת הדואר האלקטרוני שלך עבור כתבות מ {author.name}
        </AriaDescription>
        {children}
      </button>
    );
  }
}

AlertsButton.propTypes = {
  ...alertButtonPropTypes,
  /**
   * CSS class names provided by Fela
   */
  className: PropTypes.string,
};

AlertsButton.defaultProps = {
  ...alertsButtonDefaultProps,
  className: null,
};

export default AlertsButton;
export { alertButtonPropTypes, alertsButtonDefaultProps, };
