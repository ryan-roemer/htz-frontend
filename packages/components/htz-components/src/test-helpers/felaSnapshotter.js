import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
