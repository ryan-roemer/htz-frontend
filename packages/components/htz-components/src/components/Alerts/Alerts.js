import React, { Fragment, } from 'react';
import { FelaComponent, } from 'react-fela';
import Media from '../Media/Media';
import AlertsButton, { authorPropTypes, } from './AlertsButton';
import IconMail from '../Icon/icons/IconMail';
import IconAlert from '../Icon/icons/IconAlert';

function Alerts({ author, }) {
  return (
    <FelaComponent
      style={theme => ({
        extend: [
          theme.mq(
            { from: 's', },
            {
              display: 'block',
              width: '100%',
              textAlign: 'end',
              color: theme.color('tertiary'),
              fontWeight: 700,
              ...theme.type(-1),
            }
          ),
          theme.mq(
            { until: 's', },
            {
              textAlign: 'center',
              width: '5rem',
              ...theme.type(-2),
            }
          ),
        ],
      })}
      render={({
        className,
        theme: {
          alertsI18n: { mobileAlertsText, desktopAlertsText, },
        },
      }) => (
        <AlertsButton className={className} author={author}>
          <Media query={{ from: 's', }} matchOnServer>
            {matches =>
              (matches ? (
                <FelaComponent
                  style={{ display: 'flex', alignItems: 'center', }}
                >
                  <IconMail size={3} miscStyles={{ marginEnd: '1rem', }} />
                  {desktopAlertsText}
                </FelaComponent>
              ) : (
                <Fragment>
                  <div>
                    <IconAlert size={3} color={[ 'primary', '+1', ]} />
                  </div>
                  {mobileAlertsText}
                </Fragment>
              ))
            }
          </Media>
        </AlertsButton>
      )}
    />
  );
}

Alerts.propTypes = authorPropTypes;

export default Alerts;
