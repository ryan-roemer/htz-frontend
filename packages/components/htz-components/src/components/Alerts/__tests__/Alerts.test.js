/** @jest-environment node */
import React from 'react';
import felaSnapshotter from '../../../test-helpers/felaSnapshotter';
import Alerts from '../Alerts';

describe('Alerts component', () => {
  it('renders correctly', () => {
    const author = {
      name: 'Avi Kaufman',
    };

    const snapshot = felaSnapshotter(<Alerts author={author} />);
    expect(snapshot).toMatchSnapshot();
  });
});
