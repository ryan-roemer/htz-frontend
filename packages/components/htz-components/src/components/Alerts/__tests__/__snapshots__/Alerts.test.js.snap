// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alerts component renders correctly 1`] = `
Object {
  "component": <div>
    <button
      aria-describedby="alerts_btn_description"
      aria-expanded={false}
      className="htz-a htz-b htz-c htz-d htz-e htz-f htz-g htz-h htz-i htz-j htz-k htz-l htz-m"
      onClick={[Function]}
    >
      <span
        className="htz-n htz-o htz-p htz-q htz-r htz-s htz-t htz-u htz-v htz-w htz-x"
        id="alerts_btn_description"
      >
        קבל התראות בתיבת הדואר האלקטרוני שלך עבור כתבות מ 
        Avi Kaufman
      </span>
      <svg
        className="htz-y htz-z htz-ab htz-ac htz-ae htz-af"
        height="1em"
        viewBox="0 0 350 256"
        width="1.3671875em"
      >
         
        <path
          d="M233 46.5l-102.9 93-101.9-93H233zm-63.9 73.8l64.7 70.4-7.4 9-65.8-71.7-30.5 27.6-30.6-27.9L34 199.3l-7.4-9L91 119.9 23 57.7v151.8h215V58.1l-68.9 62.2zM327 46.5h-89v12h89v-12zm-14 37h-75v12h75v-12zm-17 38h-58v11h58v-11zm-14 38h-44v12h44v-12zm-17 38h-27v12h27v-12z"
          fill="currentColor"
        />
      </svg>
      <span
        className="htz-ag"
      >
        התראות במייל
      </span>
      <span
        className="htz-ah htz-ai"
      >
        התראות
      </span>
    </button>
  </div>,
  "styles": ".htz-a{color:#A8001C}.htz-b{font-weight:bold}.htz-n{border:0}.htz-o{clip:rect(0 0 0 0)}.htz-p{-webkit-clip-path:inset(50%)}.htz-q{clip-path:inset(50%)}.htz-r{height:1px}.htz-s{margin:-1px}.htz-t{overflow:hidden}.htz-u{padding:0}.htz-v{position:absolute}.htz-w{width:1px}.htz-x{white-space:nowrap}.htz-y{vertical-align:-0.15em}.htz-z{color:inherit}.htz-ab{fill:#FFF}.htz-af{margin-left:1rem}.htz-ah{display:block}@media (min-width: 80em){.htz-c{font-size:2rem}.htz-d{line-height:1.5em}}@media (min-width: 64em) and (max-width: 79.9375em){.htz-e{font-size:2.3333333333333335rem}.htz-f{line-height:1.2857142857142858em}}@media (max-width: 63.9375em){.htz-g{font-size:2rem}.htz-h{line-height:1.5em}}@media (min-width: 37.5em){.htz-i{width:100%}.htz-j{text-align:left}.htz-ae{font-size:2.5rem}.htz-ai{display:none}}@media (min-width: 37.5em) and (max-width: 63.9375em){.htz-k{display:inline}}@media (min-width: 64em){.htz-l{display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex}}@media (max-width: 37.4375em){.htz-m{text-align:center}.htz-ac{font-size:3rem}.htz-ag{display:none}}",
}
`;
